server/src/index.js

```diff
@@ -1,7 +1,14 @@
+const config = require('../config');
 const app = require('./server');
+const seed = require('./utils/seed');
 
-const port = 3001;
+const port = config.api.port;
 
-app.listen(port, function() {
+// ##### for development only #####
+if (config.env === 'development') {
+  seed();
```

Comment:
Does it matter if this is an async function that you might start listening before the db is seeded?

Reply:
Yep, it matters. I thought about that when I wrote it. I even tested it last week by trying to start the app and get a request through on Postman before the database could be seeded. I could never beat it!

Technically, this is not a complete solution for that reason. Practically, because this is only for development and once your database is in good shape it takes 0 time to run, it is probably not worth implementing a hairier solution to account for the situation that you make a request before the tables are created the first time you open the app.

Reply:
Makes sense. I guess we could add some error handling where the server won't start if the `seed` fails? Although I guess it's not dependent on there being data...but it is kinda dependent on there being tables. But the error would be handled through the failing API call. Hm 🤔

Reply:
Yeah it is dependent on the tables being there, but it is also dependent on the database running, etc. If the table-create fails, then the database is probably acting up and there will probably some connection errors there, which are handled by the db driver (i think)